### ISSUE TITLE
Code formatting for Leo responses

### ISSUE THREAD

**Issue by petemill** [brave/brave-browser#33668]
Identify which parts of Leo's response is code, and format appropriately.

<img width="281" alt="image" src="https://github.com/brave/brave-browser/assets/741836/7e3014a2-91d0-4c82-bec1-a24fd80902a3">

Figma: https://www.figma.com/file/MNwi0uwXNhkAbPiJ86G4vm/%F0%9F%A7%A0-Leo?type=design&node-id=1592-7938&mode=design&t=mmPIf91aKWeWk2PA-0

**Comment by petemill** [comment]
@LorenzoMinto do you have any preferences?

**Comment by LorenzoMinto** [comment]
Codellama seems to have been trained both with \`\`\`, \`\`\`{language} or even `[{language}][/{language}]` tags (similar to the instruction ones). I guess we could use the traditional \`\`\` (or \`\`\`{language}) also to be consistent with what we are going to expect from other models

**Comment by thypon** [comment]
Are we planning/using markdown anywhere there?

**Comment by kjozwiak** [comment]
Adding `QA/Blocked` for now as the above was merged into `cr121-1.62.x` via https://github.com/brave/brave-core/pull/21593. Once `C121` is merged into `1.62.x` via https://github.com/brave/brave-core/pull/21593, we'll unblock the above and mention the needed `1.62.x` version for verification 👍

**Comment by kjozwiak** [comment]
The above requires `1.62.148` or higher for `1.62.x` verification 👍

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

```
Brave | 1.62.149 Chromium: 121.0.6167.75 (Official Build) (x86_64)
-- | --
Revision | ff84587bd70af9fcbcbe59fc5194ca65082759c4
OS | macOS Version 11.7.10 (Build 20G1427)
```
